rcl/include/rcl/log_level.h

```diff
@@ -0,0 +1,74 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCL__LOG_LEVEL_H_
+#define RCL__LOG_LEVEL_H_
+
+#include "rcl/allocator.h"
+#include "rcl/macros.h"
+#include "rcl/types.h"
+#include "rcl/visibility_control.h"
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+/// A logger item to specify a name and log_level
+typedef struct rcl_logger_setting_t
+{
+  /// name for the logger.
+  char * name;
+  /// level for the logger.
+  int level;
+} rcl_logger_setting_t;
+
+/// Hold log_level.
+typedef struct rcl_log_level_t
+{
+  /// Default log level (represented by `RCUTILS_LOG_SEVERITY` enum) or -1 if not specified.
+  int default_log_level;
+  struct rcl_logger_setting_t * logger_settings;   ///<  Array of logger
+  size_t num_loggers;       ///< Number of loggers
+  rcl_allocator_t allocator;
+} rcl_log_level_t;
+
+
+/// \brief Initialize log level structure
+/// \param[in] allocator memory allocator to be used
+/// \return a pointer to log level structure on success or NULL on failure
+RCL_PUBLIC
+RCL_WARN_UNUSED
+rcl_log_level_t *
+rcl_log_level_init(const rcutils_allocator_t allocator);
```

Comment:
I would recommend having two functions:
```c
rcl_log_level_t rcl_get_zero_initialized_log_level();

rcl_ret_t
rcl_log_level_init(rcl_log_level_t * log_level, const rcl_allocator_t * allocator);
```

This approach allows having a stack allocated `rcl_log_level_t` (of course, it's member would still be heap allocated) and better error handling in the init function.

Reply:
Notes:
- Use `rcl_allocator_t` and not `rcutils_allocator_t`.
- We have some functions passing allocators by value, and some others by value. We have more of the first kind so I would like to standardize to that style.

Reply:
It needs a variable with heap allocated here.
```
typedef struct rcl_arguments_impl_t
...
  rcl_log_level_t * log_level;                // not define with `rcl_log_level_t log_level;`
}

args_impl->log_level = rcl_log_level_init(allocator);
```

use the same style referred to 
https://github.com/ros2/rcl/blob/b9cfc243bf018091b5c5bcb5585d19591daf4ae6/rcl/src/rcl/arguments.c#L280

if updating with your recommendation, rcl_log_level_t rcl_get_zero_initialized_log_level seems not working.

Reply:
You could still heap allocate the struct:

```c
args_impl->log_levels = malloc(sizeof(rcl_log_levels_t));
if (NULL == args_impl->log_levels) {
  ... handle error
}
*args_impl->log_levels = rcl_get_zero_initialized_log_level();
rcl_ret_t ret = rcl_log_level_init(args_impl->log_levels, allocator);
...
```

Considering that previous API was already written like this https://github.com/ros2/rcl/pull/664#discussion_r447684454, I think it's fine to keep the same style.